pkg/identity/identitymanager/manager.go

```diff
@@ -93,12 +93,14 @@ func (idm *IdentityManager) add(identity *identity.Identity) {
 			identity: identity,
 			refCount: 1,
 		}
+		identity.ReferenceCount = 1
 		for o := range idm.observers {
 			o.LocalEndpointIdentityAdded(identity)
 		}
 
 	} else {
 		idMeta.refCount++
+		identity.ReferenceCount = idMeta.refCount
```

Comment:
Not sure how these fixes to `identity.ReferenceCount` play with the reference counting for local identities in `pkg/identity/cache/local.go`. Likely there is no effect as local identities are only used for CIDRs and these are endpoint identities? (@joestringer)

Reply:
If we end up keeping these `identity.ReferenceCount` updates in here, the locking rules to protect against concurrent updates should be documented.